processing/src/main/java/org/apache/druid/segment/column/BaseColumn.java

```diff
@@ -41,4 +42,11 @@ default VectorObjectSelector makeVectorObjectSelector(ReadableVectorOffset offse
   {
     throw new UOE("Cannot make VectorObjectSelector for column with class[%s]", getClass().getName());
   }
+
+  @SuppressWarnings("unused")
+  @Nullable
+  default <T> T as(Class<? extends T> clazz)
+  {
+    return null;
```

Comment:
The intention is to have implementations of base column be classes that have semantic creators and have possible conversions. Overrides can be registered against them. 
If a class does not have any conversion to another class, and does not override this function, it defaults to null. 

Adding a map and allowing functions to register overrides to it may cause it to use the conversions from the base class, if it is not overridden. I am not sure if that behavior is expected.